test/e2e/specs/editor/various/preview.spec.js

```diff
@@ -0,0 +1,362 @@
+		const previewTitle = previewPage.locator( 'role=heading[level=1]' );
+		expect( await previewTitle.textContent() ).toEqual( 'Lorem' );
+
+		// Return to editor and publish post.
+		await editorPage.bringToFront();
+		await editor.publishPost();
+
+		// Close the panel.
+		await page.locator( 'role=button[name="Close panel"]' ).click();
+
+		// Change the title and preview again.
+		await editorPage.type( '[aria-label="Add title"]', ' Ipsum' );
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match updated input.
+		expect( await previewTitle.textContent() ).toEqual( 'Lorem Ipsum' );
+
+		// Return to editor and switch to Draft.
+		await editorPage.bringToFront();
+		await page.locator( 'role=button[name="Switch to draft"]' ).click();
+
+		// Change the title.
+		await editorPage.type( '[aria-label="Add title"]', ' Draft' );
+
+		// Open the preview page.
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match updated input.
+		expect( await previewTitle.textContent() ).toEqual(
+			'Lorem Ipsum Draft'
+		);
+
+		await previewPage.close();
+	} );
+} );
+
+test.describe( 'Preview with Custom Fields enabled', () => {
+	test.beforeEach( async ( { admin, previewUtils } ) => {
+		await admin.createNewPost();
+		await previewUtils.toggleCustomFieldsOption( true );
+	} );
+
+	test.afterEach( async ( { previewUtils } ) => {
+		await previewUtils.toggleCustomFieldsOption( false );
+	} );
+
+	// Catch regressions of https://github.com/WordPress/gutenberg/issues/12617
+	test( 'displays edits to the post title and content in the preview', async ( {
+		editor,
+		page,
+		previewUtils,
+	} ) => {
+		const editorPage = page;
+
+		// Add an initial title and content.
+		await editorPage.type( 'role=textbox[name="Add title"i]', 'title 1' );
+		await editorPage.type(
+			'role=button[name="Add default block"i]',
+			'content 1'
+		);
+
+		// Publish the post and then close the publish panel.
+		await editor.publishPost();
+
+		// Close the panel.
+		await page.locator( 'role=button[name="Close panel"]' ).click();
+
+		// Open the preview page.
+		const previewPage = await editor.openPreviewPage();
+		await previewPage.waitForSelector( 'role=heading[level=1]' );
+
+		// Check the title and preview match.
+		const previewTitle = previewPage.locator( 'role=heading[level=1]' );
+		expect( await previewTitle.textContent() ).toEqual( 'title 1' );
+		// No semantics we can grab here; it's just a <p> inside a <div> :')
+		const previewContent = previewPage.locator( '.entry-content p' );
+		expect( await previewContent.textContent() ).toEqual( 'content 1' );
+
+		// Return to editor and modify the title and content.
+		await editorPage.bringToFront();
+		await editorPage.click( 'role=textbox[name="Add title"i]' );
+		await editorPage.keyboard.press( 'End' );
+		await editorPage.keyboard.press( 'Backspace' );
+		await editorPage.keyboard.type( '2' );
```

Comment:
Instead of doing this _manually_. We can try using `.fill()` to make it more readable.

```suggestion
		await editorPage.fill( 'role=textbox[name="Add title"i]', 'title 2' )
```